src/gas-price/gas-price.ts

```diff
@@ -81,9 +81,20 @@ export const fetchAndStoreGasPrice = async (
   return gasPrice;
 };
 
-export const getRecommendedGasPrice = (chainId: string, providerName: string, sponsorWalletAddress: Address) => {
+export const getRecommendedGasPrice = (
+  chainId: string,
+  providerName: string,
+  sponsorWalletAddress: Address,
+  dataFeedIds: Hex[]
+) => {
   const state = getState();
-  const pendingTransactionInfo = state.pendingTransactionsInfo[chainId]![providerName]![sponsorWalletAddress];
+  const oldestPendingTransactionInfo = minBy(
+    dataFeedIds
+      .map((dataFeedId) => state.pendingTransactionsInfo[chainId]?.[providerName]?.[sponsorWalletAddress]?.[dataFeedId])
+      .sort((a, b) => (b?.consecutivelyUpdatableCount ?? 0) - (a?.consecutivelyUpdatableCount ?? 0)), // Sort by consecutivelyUpdatableCount
```

Comment:
So this is a similar comment to the one @bdrhn9 made [here](https://github.com/api3dao/airseeker-v2/pull/296#discussion_r1593936792). I ended up sorting by  `consecutivelyUpdatableCount` because each feed in a batch will have the same [firstUpdatableTimestamp](https://github.com/api3dao/airseeker-v2/blob/single-sponsor-wallet/src/update-feeds-loops/pending-transaction-info.ts#L45) therefore I made the assumption that the data feed that has been retried the most should be oldest. Maybe I should sort by both properties 🤔

Reply:
Nah, just sort by the `firstUpdatableTimestamp`. The `consecutivelyUpdatableCount` is really only for infomational purposes (and maybe later for alerting). But even better, as Bedirhan pointed out - do not sort at all, because you use `minBy` anyway.